enhancements/service-binding.md

```diff
@@ -0,0 +1,291 @@
+---
+title: service-binding
+authors:
+  - "@sbose78"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-03-19
+last-updated: 2021-03-03
+status: proposed
+---
+
+# Service Binding for operator-backed services
```

Comment:
very little of this seems specific to operators.  it's generically about binding for a variety of resource types.

1) how much of this represents capabilities the SBO already has?
2) what does it really have to do with OLM?  (at a minimum i'd expect to see examples/discussion of how operators expose bindable information, or bind to information, using the CSV metadata mechanism).

Reply:
> how much of this represents capabilities the SBO already has?

All of it


> what does it really have to do with OLM? (at a minimum i'd expect to see examples/discussion of how operators expose bindable information, or bind to information, using the CSV metadata mechanism).

Yes, CSV descriptors are an option. I wanted to cast a wider net with Operators/CRDs in this proposal. Let me add in the pieces for CSV too.

Reply:
> All of it

so what is the enhancement that is being proposed?  this reads more like (possibly sorely needed) product docs.  Or preamble/background leading up to a proposal but not actually making one.

Reply:
> so what is the enhancement that is being proposed? 
A proposal to enhance the Service Binding Operator would be made to the Service Binding Operator project. 

This proposal describes what the Service Binding Operator is, with the ultimate goal of making it an Operator Framework project ( ie, a part of it). The latter is what I'm really proposing.

Reply:
> This proposal describes what the Service Binding Operator is, with the ultimate goal of making it an Operator Framework project ( ie, a part of it). The latter is what I'm really proposing.

ok, can you clarify what that would mean?  (lift+shift the operator into the operator-framework github org?  deprecate the standalone operator and embed the binding logic/behavior into OLM itself?  something else?)  That seems like the critical piece of the proposal.  (But the background is very helpful)